ui/src/combobox.tsx

```diff
@@ -59,54 +60,108 @@ export interface Combobox {
   required?: B
 }
 
-export const
-  XCombobox = ({ model: m }: { model: Combobox }) => {
-    const
-      isMultiValued = !!m.values,
-      [text, setText] = React.useState(m.value),
-      [options, setOptions] = React.useState((m.choices || []).map((text): Fluent.IComboBoxOption => ({ key: text, text }))),
-      [selected, setSelected] = React.useState<Fluent.IComboBoxOption[]>((m.values || []).map(text => ({ key: text, text }))),
-      selectOpt = (option: Fluent.IComboBoxOption) => {
-        setSelected(items => {
-          const result = option.selected ? [...items, option] : items.filter(item => item.key !== option.key)
-          wave.args[m.name] = result.map(item => item.text)
-          return result
-        })
-      },
-      onChange = (_e: React.FormEvent<Fluent.IComboBox>, option?: Fluent.IComboBoxOption, _index?: U, value?: S) => {
-        if (!option && value) {
-          const opt: Fluent.IComboBoxOption = { key: value, text: value }
-          setOptions((prevOptions = []) => [...prevOptions, opt])
-          selectOpt({...opt, selected: true})
-        }
-        if (option && isMultiValued) {
-          selectOpt(option)
-        }
-        else {
-          const v = option?.text || value || ''
-          wave.args[m.name] = v
-          setText(v)
-        }
-        if (m.trigger) wave.push()
+export const XCombobox = ({ model: m }: { model: Combobox }) => !m.values ? <ComboboxSingleSelect model={m} /> : <ComboboxMultiSelect model={m} />
```

Comment:
Negated conditions are a bit harder to read in general so let's flip it
```suggestion
export const XCombobox = ({ model: m }: { model: Combobox }) => m.values ? <ComboboxMultiSelect model={m} /> : <ComboboxSingleSelect model={m} />
```